KnowledgeExtractor/WikipediaKnowledgeGraphExtractor.cs

```diff
@@ -96,34 +101,70 @@ public static KnowledgeGraph ExtractKnGraphFromUris(List<UriAndOriginalGraphType
         {
             KnowledgeGraph result = new KnowledgeGraph();
-
-            // get the kn graph from the uris
             result = ExtractKnGraphFromUris(WikipediaPagesToParse);
+            AddEdgesBetweenAlgorithmsAndDataStructures(ref result);
+            return result;
+        }
+
+        public static void AddEdgesBetweenAlgorithmsAndDataStructures(ref KnowledgeGraph graph)
+        {
+            List<WordCount> dataStructureWordsInGraph = GetDataStructureWordsInEachWikiPage();
+            Dictionary<string, KnGNode> dataStructureNodes = GetDataStructureNodesWithoutDuplicates(graph);
+            
+            foreach (var dataStructureWordsToAddInOneNode in dataStructureWordsInGraph)
+            {
+                AddDataStructureWordsToNode(graph: ref graph, wordsToAddToNode: dataStructureWordsToAddInOneNode, dataStructureNodes: dataStructureNodes);
+            }
+        }
 
-            // add the additional edges from the data structures word count to show what data structures each algorithm and data structure is using
-            // get the word counts from the json file
-            List<WordCount> dsWordsForGraph = JsonConvert.DeserializeObject<List<WordCount>>(File.ReadAllText("../../../dataStructureWordsCountForNodesInGraph.json"));
-            // get the nodes for the DS words
-            List<KnGNode> dataStructureNodes = result.KnGraph.Where(n => n.OriginalGraphType == OriginalGraphType.DataStructuresKnGraph).ToList();
-            List<string> dsWordsToIgnore = new List<string>() { "other", "union", "reference", "arrays", "image", "matrix", "integer", "character", "list", "stack", "queue", "record", "bitmap", "set" };
-            // add the additional edges to the graph according to the json file
-            foreach (var dsWordsForNode in dsWordsForGraph)
+        private static void AddDataStructureWordsToNode(ref KnowledgeGraph graph, WordCount wordsToAddToNode, Dictionary<string, KnGNode> dataStructureNodes)
+        {
+            List<string> words = GetWords(wordsToAddToNode);
+            List<string> filteredWords = GetFilteredWords(words, DataStructureWordsToIgnore);
+
+            foreach (var word in filteredWords)
             {
-                int nodeIndex = dsWordsForNode.Index;
-                List<string> dsWords = dsWordsForNode.WordsCount.Keys.ToList();
-                dsWords = dsWords.Select(s => s.ToLowerInvariant()).ToList();
-                List<string> dsWordsFiltered = dsWords.Except(dsWordsToIgnore).ToList();
+                AddEdgeToKnowledgeGraph(graph: ref graph, nodeIndex: wordsToAddToNode.Index, nodeToAdd: dataStructureNodes[word]);
+            }
+        }
 
-                foreach (var dsWord in dsWordsFiltered)
+        private static List<string> GetFilteredWords(List<string> words, List<string> wordsToIgnore)
+        {
+            return words.Except(wordsToIgnore).ToList();
+        }
+
+        private static List<string> GetWords(WordCount wordCount)
```

Comment:
WordCount is a somewhat misleading name for the data structure here. To me 'WordCount' implies a single integer, which is a count of words. This is clearly something more complex than that. Assuming it is a count for each word, I'd call it something like WordCountByWord to make that more obvious